Lib/pydoc.py

```diff
@@ -1500,13 +1500,17 @@ def docroutine(self, object, name=None, mod=None, cl=None):
         argspec = None
 
         if inspect.isroutine(object):
+            if realname == "<lambda>":
+                retann = object.__annotations__.pop("return", None)
+            else:
+                retann = None
             argspec = _getargspec(object)
             if argspec and realname == '<lambda>':
                 title = self.bold(name) + ' lambda '
-                # XXX lambda's won't usually have func_annotations['return']
-                # since the syntax doesn't support but it is possible.
-                # So removing parentheses isn't truly safe.
-                argspec = argspec[1:-1] # remove parentheses
+                argspec = argspec[1:-1]
+                if retann is not None:
+                    argspec += f' -> {inspect.formatannotation(retann)}'
+                    object.__annotations__['return'] = retann
```

Comment:
This line is needed to restore `__annotations__` of object (this also makes `help(lambda_with_annotations)` reusable)